packages/next/src/client/components/react-dev-overlay/internal/container/Errors.tsx

```diff
@@ -307,7 +307,9 @@ export function Errors({
               {/* If there's hydration warning, skip displaying the error name */}
               {hydrationWarning ? '' : error.name + ': '}
               <HotlinkedText
-                text={hydrationWarning || error.message}
+                text={
+                  isAppDir ? hydrationWarning || error.message : error.message
```

Comment:
Reverts the `Errors.tsx` changes from:
- https://github.com/vercel/next.js/pull/65276

to match the React 18 behavior when not in app dir